main/src/cgeo/geocaching/calculator/FormulaParser.java

```diff
@@ -0,0 +1,379 @@
+                }
+
+                final String lon = parseGroups(matcher, partCount + 1, groupCount);
+                if (lon == null || lon.isEmpty()) {
+                    return null;
+                }
+
+                return new ImmutablePair<>(lat, lon);
+            }
+
+            return null;
+        }
+
+        /**
+         * @see AbstractFormulaParser#parse(String, Geopoint.LatLon)
+         */
+        @Override
+        @Nullable
+        public final ResultWrapper parse(@NonNull final String text, @NonNull final Geopoint.LatLon latlon) {
+            final MatcherWrapper matcher = new MatcherWrapper(latlon == Geopoint.LatLon.LAT ? latPattern : lonPattern, text);
+            if (matcher.find()) {
+                final String res = parseGroups(matcher, 1, matcher.groupCount());
+                if (res != null) {
+                    return new ResultWrapper(res, matcher.group().length());
+                }
+            }
+
+            return null;
+        }
+
+        /**
+         * Parses latitude or longitude from matched groups of corresponding pattern.
+         *
+         * @param groups
+         *            the groups matched by latitude/longitude pattern
+         * @return parsed latitude/longitude, or null if parsing failed
+         */
+        @Nullable
+        public abstract String parse(@NonNull List<String> groups);
+    }
+
+
+    /**
+     * Parser for MinDec format: X DD° MM.MMM'.
+     */
+    private static final class MinDecFormulaParser extends AbstractLatLonFormulaParser {
+
+        private static final String COORD_FORMULA_PATTERN_STRING = "[\\[\\]\\(\\){}" + CalculationUtils.VALID_OPERATOR_PATTERN + "A-Za-z\\d]+";
+        private static final String STRING_MINDEC = "\\s*(" + COORD_FORMULA_PATTERN_STRING + ")[°\\s]+(" + COORD_FORMULA_PATTERN_STRING + "\\." + COORD_FORMULA_PATTERN_STRING + ")['′\\s]?";
+
+        // private static final String FORMULA_PATTERN_STRING = "\\s*[NS]" + COORD_FORMULA_PATTERN_STRING + "\\s*[EW]" + COORD_FORMULA_PATTERN_STRING ;
+        // private static final Pattern FORMULA_PATTERN = Pattern.compile(FORMULA_PATTERN_STRING);
+
+
+
+        //                                           (  1  )    (    2    )    (      3      )
+        private static final String STRING_LAT = "([NS]?)" + STRING_MINDEC;
+
+        //                                        (   1  )    (    2    )    (      3      )
+        private static final String STRING_LON = "([WEO]?)" + STRING_MINDEC;
+        private static final String STRING_SEPARATOR = "[^\\w'′\"″°." + WPC_DELIM + "]*";
+        private static final Pattern PATTERN_LAT = Pattern.compile(STRING_LAT, Pattern.CASE_INSENSITIVE);
+        private static final Pattern PATTERN_LON = Pattern.compile("\\b" + STRING_LON, Pattern.CASE_INSENSITIVE);
+        private static final Pattern PATTERN_LATLON = Pattern.compile(STRING_LAT + STRING_SEPARATOR + STRING_LON, Pattern.CASE_INSENSITIVE);
+
+        MinDecFormulaParser() {
```

Comment:
There should be coordinate parsers for many coordinates formats in some other class available (alas, they don't parse for variable values inside of course). Can't they be reused somehow (parametereized to accept also letters instead of numbers)?

Reply:
Nice idea, didn't thougt about that. I will have a look into that...

Reply:
I looked into it and remembered that I thought about it, but the other parser returns a Point /Double, whereas I need a string. So I just used it to copy and modify.
But I will think, if probably it can be reused somehow else, or just extend the classes with methods returning a string, which then can be converted to double ...

Reply:
If its too complicated don't do it. I assume the formulas only work for MicDec-formatted coordinates anyway, do they?

Reply:
You are probably right, for the other formats only letters will be accepted instead of digits, but no formula.
Formulas there only can appear in the variables.

Reply:
Did not manage yet to extract the logic. 

I will keep the AbstractParser for the other coordinate-formats, if there will be also parsers for it later